File: examples/governance_static_go/compliance_static_go/domain/opened_review.go
package domain

import "time"

type OpenedReview struct {
	AggregateID string    `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	ModelId string `json:"model_id"`
	PolicyId string `json:"policy_id"`
	ReviewerId string `json:"reviewer_id"`
	Outcome string `json:"outcome"`
	Notes string `json:"notes"`
	CompletedAt time.Time `json:"completed_at"`
	Conditions []ReviewCondition `json:"conditions"`
	Status string `json:"status"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (e OpenedReview) EventName() string { return "OpenedReview" }

func (e OpenedReview) GetOccurredAt() time.Time { return e.OccurredAt }

File: examples/governance/compliance_domain/lib/compliance_domain/compliance_review/events/opened_review.rb
module ComplianceDomain
  class ComplianceReview
    module Events
      class OpenedReview
        attr_reader :aggregate_id, :model_id, :policy_id, :reviewer_id, :outcome, :notes, :completed_at, :conditions, :status, :occurred_at

        def initialize(aggregate_id: nil, model_id: nil, policy_id: nil, reviewer_id: nil, outcome: nil, notes: nil, completed_at: nil, conditions: nil, status: nil)
          @aggregate_id = aggregate_id
          @model_id = model_id
          @policy_id = policy_id
          @reviewer_id = reviewer_id
          @outcome = outcome
          @notes = notes
          @completed_at = completed_at
          @conditions = conditions
          @status = status
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end
